File: src/main/java/com/dddrivebye/usermanagement/domain/event/DriverProfileValidatedEvent.java
package com.dddrivebye.usermanagement.domain.event;

import com.dddrivebye.shared.domain.event.BaseDomainEvent;
import com.dddrivebye.usermanagement.domain.valueobject.UserId;

public final class DriverProfileValidatedEvent extends BaseDomainEvent {

    private final UserId userId;

    public DriverProfileValidatedEvent(UserId userId) {
        this.userId = userId;
    }

    public UserId userId() {
        return userId;
    }
}

File: src/modules/user-management/domain/events/driver-profile-validated.event.ts
import { BaseDomainEvent } from '@shared/domain/events/base-domain-event';

export class DriverProfileValidatedEvent extends BaseDomainEvent {
  public readonly eventName = 'driver-profile.validated';
  constructor(
    public readonly driverProfileId: string,
    public readonly userId: string,
  ) { super(); }
}
